content/blog/2020-08-14-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,242 @@
+
+For a direct comparison, we can see how this process works in spreadsheets, manipulating our data in an Excel workbook.
+
+![Nested logic](screenshots/nested-logic.png)
+
+It's a bit difficult to see the formula bar in the image above, so the formula to create used for creating the quarter column is provided below.
+
+```{}
+=IF(OR(A2="Jan",A2="Feb",A2="Mar"),"Q1",IF(OR(A2="Apr",A2="May",A2="Jun"),"Q2",IF(OR(A2="Jul",A2="Aug",A2="Sep"),"Q3",IF(OR(A2="Oct",A2="Nov",A2="Dec"),"Q4","NA"))))
+```
+
+
+## Pivot to this section
+
+Pivot tables are a powerful tool spreadsheets offer, allowing us to generate summaries of large quantities of data. Using some data wrangling functions from the tidyverse we can create summary tables as well. Let's take a look total expenses and budget by expense type and quarter.  
+
+```{r}
+spending_table <- spending_quarter %>%
+  group_by(expense, quarter) %>%
+  summarise(
+    total_expenses = sum(num_of_expenses),
+    total_expense_amt = sum(amount),
+    total_budget_amt = sum(budget),
+    .groups = "drop"
+    )
+
+spending_table
+```
+
+Again as a direct comparison, let's also take a look at what this process would look like in Excel.
+
+First we'll create a separate sheet for our pivot table, specifying the range of the data we want it to be based on. 
+
+![Pivot set up](screenshots/pivot-set-up.png)
+
+And then, it's as simple as dragging and dropping our variables of interest into specific order with `expense` and `quarter` in rows and `budget` and `amount` in values, specifying a *sum* calculation.
+
+![Pivot table](screenshots/pivot-table.png)
+
+
+## Row calculations
+
+One thing that can be tricky when working with pivot tables is calculating new variables by rows as opposed to columns. Coming back to our finance example, let's say we want to examine how much money is left over from each quarter based off of budget and amount spent.  A quick work around could be to manually add in a formula in a cell adjacent to our table and drag it down to each respective row, a process depicted below.
+
+![Left over](screenshots/left-over-calculation.png)
+
+Working with our data in R we can take advantage of the `rowwise()` function from dplyr, telling R we want to compute new values and summary statistics by row and not column.  
+Let's take a look at how much money was left over in each quarter by subtracting money spent from our budget and use the `rowwise()` function to specify we want this done by each row in our summary table.
+
+```{r}
+left_over <- spending_table %>%
+  rowwise() %>%
```

Comment:
As discussed before/elsewhere, I really feel like we can't leave this in here. `rowwise()` isn't actually doing anything and I think it will confuse people as to the actual nature of what `rowwise()` is good for.

`rowwise()` is not necessary here because R's element-wise handling of `x - y` is all the we actually need.

``` r
library(tidyverse)

dat <- tibble(x = 1:3, y = 2:4)

dat %>% 
  mutate(left_over = y - x)
#> # A tibble: 3 x 3
#>       x     y left_over
#>   <int> <int>     <int>
#> 1     1     2         1
#> 2     2     3         1
#> 3     3     4         1

# rowwise() doesn't change the computation; we get the same result because
# native vectorization is enough
dat %>% 
  rowwise() %>% 
  mutate(left_over = y - x)
#> # A tibble: 3 x 3
#> # Rowwise: 
#>       x     y left_over
#>   <int> <int>     <int>
#> 1     1     2         1
#> 2     2     3         1
#> 3     3     4         1
```

<sup>Created on 2020-08-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

 If we want to include `rowwise()`, we must come up with an example where it's actually necessary. At this point, the most pragmatic thing to do is to drop this particular example. Alternatively, the example can be reworked to point out the native vectorization in R that requires filling / dragging in Excel.

Reply:
Removed